conda/recipes/meta.yaml

```diff
@@ -41,7 +41,6 @@ requirements:
     - {{ compiler('rust') }}
     - cargo-bundle-licenses
     - maturin >=0.15,<0.16
-    - libprotobuf =3
```

Comment:
Substrait can still be supported in a conda environment. A user would just use `maturin develop/build --features protoc` however instead of having `libprotobuf` being a pinned dependency. We are learning that having these pinned dependencies to more native libraries like this often resulting in conda installation failures due to conflicts with popular libraries like aws libraries and it just seems safer to take that approach. However, if we want to take that risk of potential conflicts we could leave `libprotobuf` in the conda build file. I really don't have much of an opinion either way on it.

Reply:
Maybe it would make sense to split this PR out into changes affecting the Python and Rust package outputs? From dask-sql's perspective we're really only worried about the crate, where it's more straightforward to opt in or out of features, but the Python package landscape is a little different and I'm not really familiar with the consumers of the conda-forge/PyPI package to know if they're using the Substrait components

Reply:
Maybe it does make sense to just leave `libprotobuf` in there for now and if we run into any issues down the road make the change. I will revert this change now.